openmc_plotter/plotmodel.py

```diff
@@ -735,110 +764,109 @@ class PlotView(openmc.lib.plot._PlotBase):
-        self.outlines = False
-        self.colormaps = {'temperature': 'Oranges', 'density': 'Greys'}
-        # set defaults for color dialog
-        self.data_minmax = {prop: (0.0, 0.0) for prop in _MODEL_PROPERTIES}
-        self.user_minmax = {prop: (0.0, 0.0) for prop in _MODEL_PROPERTIES}
-        self.use_custom_minmax = {prop: False for prop in _MODEL_PROPERTIES}
-        self.data_indicator_enabled = {prop: False for prop in _MODEL_PROPERTIES}
-        self.color_scale_log = {prop: False for prop in _MODEL_PROPERTIES}
-        # Get model domain info
-        self.cells = self.getDomains('cell')
-        self.materials = self.getDomains('material')
-
-        # Tally Viz Settings
-        self.tallyDataColormap = 'spectral'
-        self.tallyDataVisible = True
-        self.tallyDataAlpha = 1.0
-        self.tallyDataIndicator = False
-        self.tallyDataUserMinMax = False
-        self.tallyDataMin = 0.0
-        self.tallyDataMax = np.inf
-        self.tallyDataLogScale = False
-        self.tallyMaskZeroValues = False
-        self.clipTallyData = False
-        self.tallyValue = "Mean"
-        self.tallyContours = False
-        self.tallyContourLevels = ""
-        self.selectedTally = None
-
-    def __hash__(self):
-        return hash(self.__dict__.__str__() + self.__str__())
-
-    @staticmethod
-    def getDomains(domain_type):
-        """ Return dictionary of domain settings.
-
-        Retrieve cell or material ID numbers and names from .xml files
-        and convert to DomainView instances with default view settings.
-
-        Parameters
-        ----------
-        domain_type : {'cell', 'material'}
-            Type of domain to retrieve for dictionary
-
-        Returns
-        -------
-        domains : Dictionary of DomainView instances
-            Dictionary of cell/material DomainView instances keyed by ID
-        """
-
-        if domain_type not in ('cell', 'material'):
-            raise ValueError("Domain type, {}, requested is neither "
-                             "'cell' nor 'material'.".format(domain_type))
-
-        lib_domain = None
-        if domain_type == 'cell':
-            lib_domain = openmc.lib.cells
-        elif domain_type == 'material':
-            lib_domain = openmc.lib.materials
-
-        domains = {}
-        for domain, domain_obj in lib_domain.items():
-            name = domain_obj.name
-            domains[domain] = DomainView(domain, name, random_rgb())
-
-        # always add void to a material domain at the end
-        if domain_type == 'material':
-            void_id = _VOID_REGION
-            domains[void_id] = DomainView(void_id, "VOID",
-                                          (255, 255, 255),
-                                          False,
-                                          False)
+        if restore_view is not None:
+            # set parameters based on supplied view
+
+            # View Parameters
+            self.level = restore_view.level
```

Comment:
There are some ways we could simplify this. One would be something like:
```Python
attributes = ['level', 'origin', ...]
for attr in attributes:
    setattr(self, attr, getattr(restore_view, attr))
```
However, we might be able to simplify it even more by just copying the data within `__dict__`. Something like:
```Python
self.__dict__ = restore_view.__dict__.copy()
```
I don't know if that would set more attributes than we want though?

Reply:
Perhaps a better idea here is to use [composition instead of inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance). That is, the `PlotView` class could just have an attribute that is of type `PlotViewIndependent` rather than _inheriting_ from it. Setting the model-indepenent view would then be as simple as setting a single attribute.